src/swagger/specGenerator.ts

```diff
@@ -11,6 +11,44 @@ export abstract class SpecGenerator {
     return this.getSwaggerType(type);
   }
 
+  private buildDefinitions() {
```

Comment:
I believe that you meant to make this change to the child (`specGenerator2.ts`) and not to the parent (`specGenerator.ts`).

So please remove `specGenerator.ts`.`buildDefinition` and make those changes to the two children:
* `specGenerator2.ts`.`buildDefinition`
* `specGenerator3.ts`.`buildComponents`
  * Side note: judging by the fact nothing in `specGenerator3` calls `buildDefinitions` that tells me that the PR as it currently stands does not actually allow numerical enums for OpenAPI 3.0. Can you please add tests for the swagger definitions?
    * test file for OpenAPI 2.0 (i.e. Swagger) definitions: `tests\unit\swagger\definitionsGeneration\definitions.spec.ts`
    * test file for OpenAPI 3.0 definitions: `tests\unit\swagger\schemaDetails3.spec.ts`